File: Algorithm/programmers/Level1/약수의 개수와 덧셈.py
# 단순 계산
def solution(left, right):
    answer = 0
    for i in range(left, right + 1):
        count = 0
        for j in range(1, i + 1):
            if i % j == 0:
                count += 1
        if count % 2 == 0:
            answer += i
        else:
            answer -= i
    return answer


# 최적화
import math

def solution(left, right):
    answer = 0
    for i in range(left, right + 1, 1):
        sqrt = math.sqrt(i)
        if int(sqrt) == sqrt:
            answer -= i
        else:
            answer += i

    return answer

File: Algorithm/programmers/Level1/약수의 개수와 덧셈.js
function solution(left, right) {
  let answer = 0;

  for (let i = left; i <= right; i++) {
    let num = 1;
    let mod = 1;
    while (mod <= i / 2) {
      if (i % mod === 0) num += 1;
      mod += 1;
    }
    if (num % 2 === 0) answer += i;
    else answer -= i;
  }
  return answer;
}

function solution(left, right) {
  let answer = 0;

  for (let i = left; i <= right; i++) {
    if (Number.isInteger(Math.sqrt(i))) answer -= i;
    else answer += i;
  }

  return answer;
}
